### PR TITLE
Fixed #28474 -- Made DurationField raise ValidationError for inputs that raised OverflowError.

### PR DESCRIPTION
**Pull request by srinivasreddy**
https://code.djangoproject.com/ticket/28474

### CODE REVIEW

File: django/utils/dateparse.py [hunk 1]
```diff
@@ -136,7 +137,10 @@ def parse_duration(value):
         match = iso8601_duration_re.match(value) or postgres_interval_re.match(value)
     if match:
         kw = match.groupdict()
-        days = datetime.timedelta(float(kw.pop('days', 0) or 0))
+        days = float(kw.pop('days', 0) or 0)
+        if not timedelta.min.days <= days <= timedelta.max.days:
```
Comment (timgraham) [inline django/utils/dateparse.py]: Why is this needed? Isn't it reimplementing an exception raised by `timedelta()`? No tests seem to fail with it removed.

File: tests/utils_tests/test_duration.py [hunk 2]
```diff
@@ -23,6 +23,12 @@ def test_negative(self):
         duration = datetime.timedelta(days=-1, hours=1, minutes=3, seconds=5)
         self.assertEqual(duration_string(duration), '-1 01:03:05')
 
+    def test_overflow(self):
```
Comment (timgraham) [inline tests/utils_tests/test_duration.py]: These tests are already passing without any changes so I don't think they're needed.

File: tests/forms_tests/field_tests/test_durationfield.py [hunk 3]
```diff
@@ -18,6 +19,10 @@ def test_durationfield_clean(self):
             datetime.timedelta(days=1, hours=1, minutes=15, seconds=30, milliseconds=300),
             f.clean('1 1:15:30.3')
         )
+        with self.assertRaises(ValidationError):
```
Comment (timgraham) [inline tests/forms_tests/field_tests/test_durationfield.py]: Use assertRaisesMessage to verify the message also.

File: django/forms/fields.py [hunk 4]
```diff
@@ -472,19 +473,23 @@ def strptime(self, value, format):
 class DurationField(Field):
     default_error_messages = {
         'invalid': _('Enter a valid duration.'),
+        'overflow': _('days must be between {} and {}'.format(timedelta.min.days, timedelta.max.days))
```
Comment (timgraham) [inline django/forms/fields.py]: @claudep, does this look okay for translators? I'm not sure if it allows apply localization formatting to the numbers.

File: django/forms/fields.py [hunk 5]
```diff
@@ -472,19 +473,23 @@ def strptime(self, value, format):
...
     }
 
     def prepare_value(self, value):
-        if isinstance(value, datetime.timedelta):
+        if isinstance(value, timedelta):
```
Comment (timgraham) [inline django/forms/fields.py]: I'd suggest not changing these as it's not related to the patch.

File: django/forms/fields.py (hunk 4 above)
Comment (timgraham) [inline django/forms/fields.py]: Start the sentence "Duration's number of days..." Also, add a trailing period.

**Comment by srinivasreddy** [comment]
@timgraham  Done.

File: django/forms/fields.py [hunk 6]
```diff
@@ -468,6 +469,10 @@ def strptime(self, value, format):
 class DurationField(Field):
     default_error_messages = {
         'invalid': _('Enter a valid duration.'),
+        'overflow': _('Duration\'s number of days must be between {} and {}.'.format(
```
Comment (claudep) [inline django/forms/fields.py]: For translators, it would be better to use named placeholders, so languages which need reordering can do it nicely.

**Comment by srinivasreddy** [comment]
@claudep  Done

**Comment by srinivasreddy** [comment]
@timgraham   Documentation update is done.
@claudep  place holders are inserted.

Please take a look.

**Comment by srinivasreddy** [comment]
@timgraham   What do you think now?
